content/main/docs/serverless-workflows/configuration/token-propagation.md

```diff
@@ -0,0 +1,118 @@
+This allows the user identity to be propagated to the third parties/externals services called by the workflow.
+To do so, a set of properties must be set in the workflow `application.properties` file.
+
+# Prerequisites
+* Having a Keycloak instance running with a client
+* Having RHDH with the latest version of the Orchestrator plugins
+* Having a workflow using openapi spec file to send REST requests to a service
+
+# Build
+
+When building the workflow's image, you will need to make sure the following extensions is present in the `QUARKUS_EXTENSION`:
+* io.quarkus:quarkus-oidc-client-filter # needed for propagation
+* io.quarkus:quarkus-oidc # neded for token validity check thus accessing $WORKFLOW.identity
+
+# Configuration
+## Oauth2
+1. In the openapi spec file(s) for which you want to propagate the incoming token to, write down the security scheme configured for the endpoints of interests for you. All endpoints may use the same security scheme if configured globally.
+e.g
+```
+components:
+  securitySchemes:
+    BearerToken:
+     type: oauth2
+     flows:
+       clientCredentials:
+         tokenUrl: http://<keycloak>/realms/<yourRealm>/protocol/openid-connect/token
+         scopes: {}
+     description: Bearer Token authentication
+```
+2. In the `application.properties` of your workflow, for each security scheme, adds the following:
+```
+auth-server-url=https://<keycloak>/realms/<yourRealm>
+client-id=<client ID>
+client-secret=<client secret>
+
+# Properties to check for identity, needed to use $WORKFLOW.identity within the workflow
+quarkus.oidc.auth-server-url=${auth-server-url}
+quarkus.oidc.client-id=${client-id}
+quarkus.oidc.credentials.secret=${client-secret}
+quarkus.oidc.token.header=X-Authorization-<provider>
+quarkus.oidc.token.issuer=any # needed in case the auth server url is not the same as the one configured; e.g: localhost VS the k8S service
+
+# Properties for propagation
+quarkus.oidc-client.BearerToken.auth-server-url=${auth-server-url}
+quarkus.oidc-client.BearerToken.token-path=${auth-server-url}/protocol/openid-connect/token
+quarkus.oidc-client.BearerToken.discovery-enabled=false
+quarkus.oidc-client.BearerToken.client-id=${client-id}
+quarkus.oidc-client.BearerToken.grant.type=client
+quarkus.oidc-client.BearerToken.credentials.client-secret.method=basic
+quarkus.oidc-client.BearerToken.credentials.client-secret.value=${client-secret}
+
+quarkus.openapi-generator.<spec_file_yaml_or_json>.auth.<security_scheme>.token-propagation=true
+quarkus.openapi-generator.<spec_file_yaml_or_json>.auth.<security_scheme>.header-name=X-Authorization-<provider>
+```
+With:
+* `spec_file_yaml_or_json`: the name of the spec file configured with `_` as separator. E.g: if the file name is `dumb-server.yaml` the normalized property name will be `dumb_server_yaml`. This should be the same for every security scheme defined in the file.
```

Comment:
perhaps `sample-server` instead of `dumb-server`? keeping it positive ;-)

Reply:
fine :)